automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetes/KubernetesSpec.scala

```diff
@@ -0,0 +1,243 @@
+import org.broadinstitute.dsde.workbench.leonardo.http.{GetAppResponse, ListAppResponse, PersistentDiskRequest}
+import org.http4s.{AuthScheme, Credentials}
+import org.http4s.headers.Authorization
+import org.scalatest.{DoNotDiscover, ParallelTestExecution}
+
+import scala.concurrent.duration._
+
+@DoNotDiscover
+class KubernetesSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPAllocUtils with ParallelTestExecution {
+
+  implicit val authTokenForOldApiClient = ronAuthToken
+  implicit val auth: Authorization =
+    Authorization(Credentials.Token(AuthScheme.Bearer, ronCreds.makeAuthToken().value))
+
+  val dependencies = for {
+    httpClient <- LeonardoApiClient.client
+  } yield AppDependencies(httpClient)
+
+  "create app when cluster doesn't exist" in { _ =>
+    withNewProject { googleProject =>
+      val appName = randomAppName
+      val appName2 = randomAppName
+
+      val app1DeletedDoneCheckable: DoneCheckable[List[ListAppResponse]] =
+        x => x.filter(_.appName == appName).map(_.status).distinct == List(AppStatus.Deleted)
+
+      val app2DeletedDoneCheckable: DoneCheckable[List[ListAppResponse]] =
+        x => x.filter(_.appName == appName2).map(_.status).distinct == List(AppStatus.Deleted)
+
+      logger.info(s"Google Project 1 " + googleProject.value)
+
+      val createAppRequest = defaultCreateAppRequest.copy(
+        diskConfig = Some(
+          PersistentDiskRequest(
+            randomDiskName,
+            Some(DiskSize(30)),
+            None,
+            Map.empty
+          )
+        )
+      )
+      val res = dependencies.use { dep =>
+        implicit val client = dep.httpClient
+        val creatingDoneCheckable: DoneCheckable[GetAppResponse] =
+          x => x.status == AppStatus.Running
+
+        for {
+
+          _ <- loggerIO.info("About to create app")
+
+          _ <- LeonardoApiClient.createApp(googleProject, appName, createAppRequest)
+
+          gar = LeonardoApiClient.getApp(googleProject, appName)
+
+          getAppResponse <- gar
+
+          _ = getAppResponse.status should (be(AppStatus.Provisioning) or be(AppStatus.Precreating))
+
+          monitorStartingResult <- testTimer.sleep(120 seconds) >> streamFUntilDone(gar, 30, 30 seconds)(
+            testTimer,
+            creatingDoneCheckable
+          ).compile.lastOrError
+
+          _ <- loggerIO.info(s"app 1 monitor result: ${monitorStartingResult}")
+
+          _ = monitorStartingResult.status shouldBe AppStatus.Running
+
+          _ <- LeonardoApiClient.deleteApp(googleProject, appName)
+
+          listApps = LeonardoApiClient.listApps(googleProject, true)
+
+          monitorApp1DeletionResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(listApps, 120, 10 seconds)(
+            testTimer,
+            app1DeletedDoneCheckable
+          ).compile.lastOrError
+
+          _ <- loggerIO.info(s"app1 delete result: $monitorApp1DeletionResult")
+
+          _ <- testTimer.sleep(480 seconds)
```

Comment:
Do we need to create a second app here? It adds a lot of time to the test.. is creating 1 app enough to test `"create app when cluster doesn't exist"`?